src/checklyhq-provider.ts

```diff
@@ -0,0 +1,131 @@
+/* Copyright © 2022 Seneca Project Contributors, MIT License. */
+import { ChecklyhqProviderOptions } from './types'
+import { getJSON, makeConfig, makeUrl, postJSON } from './utils'
+
+const Pkg = require('../package.json')
+
+import fetch from "node-fetch";
+
+function CheklyhqProvider(this: any, options: ChecklyhqProviderOptions) {
+  const seneca: any = this
+
+  const entityBuilder = this.export('provider/entityBuilder')
+
+  seneca.message('sys:provider,provider:checklyhq,get:info', get_info)
+
+  async function get_info(this: any, _msg: any) {
+    return {
+      ok: true,
+      name: 'checklyhq',
+      version: Pkg.version,
+    }
+  }
+
+  entityBuilder(this, {
+    provider: {
+      name: 'checklyhq'
+    },
+    entity: {
+      checks: {
+        cmd: {
+          list: {
+            action: async function(this: any, entize: any, msg: any) {
+              const res: any = await getJSON(makeUrl('checks', msg.q, options), options, makeConfig(this))
+              let checks = res
+              let list = checks.map((data: any) => entize(data))
+              
+              return list
+            },
+          },
+          load: {
+            action: async function(this: any, entize: any, msg: any) {
+              const res: any = await getJSON(makeUrl('checks', msg.q, options), options, makeConfig(this))
+              let load = res ? entize(res) : null
+
+              return load
+            },
+          },
+          save: {
+            action: async function(this: any, entize: any, msg: any) {
+              const body = this.util.deep(
+                this.shared.primary,
```

Comment:
surely name should be defined as a ent property since you will have many checks with many names?

Reply:
Indeed. I removed this line